src/puppetlabs/trapperkeeper/services/status/ringutils.clj

```diff
@@ -14,7 +14,8 @@
   [handler]
   (fn [request]
     (try+ (handler request)
-          (catch [:type :request-data-invalid] e
+          (catch #(or (= :request-data-invalid (:type %))
+                      (= :service-status-version-not-found (:type %))) e
```

Comment:
This could be `#(contains? #{:request-data-invalid :service-status-version-not-found} (:type %))`, but, that's not a blocker